js/main.js

```diff
@@ -32,6 +32,81 @@ function initNonVueButtons () {
+        'Last 30 days': [,],
+        'Last 90 days': [,],
+        'Last 6 months': [,],
+        'Last 1 year': [,],
+        'Today': [,],
+        'Yesterday': [,],
+        'This week': [,],
+        'This month': [,],
+        'Last month': [,]
+      },
+      'alwaysShowCalendars': true,
+      'startDate': moment(window.MetricQWebView.instances[0].handler.startTime),
+      'endDate': moment(window.MetricQWebView.instances[0].handler.stopTime)
+    }, function (start, end, label) {
+      if (Math.abs(moment().unix() - end.unix()) < 60 * 2) {
+        window.MetricQWebView.instances[0].relative = true
+        window.MetricQWebView.instances[0].handler.setTimeRange(start.unix() * 1000, end.unix() * 1000)
+      } else {
+        window.MetricQWebView.instances[0].relative = false
+        window.MetricQWebView.instances[0].handler.setTimeRange(start.unix() * 1000, end.unix() * 1000)
+      }
+      window.MetricQWebView.instances[0].reload()
+    })
+    $('input[name="daterange"]').on('show.daterangepicker', function (ev, picker) {
+      picker.ranges = {
+        'last 3 hours': [moment().subtract(3, 'hours'), moment()],
+        'Today': [moment().startOf('day'), moment()],
+        'Yesterday': [moment().subtract(1, 'days').startOf('day'), moment().startOf('day')],
+        'Last 7 Days': [moment().subtract(6, 'days'), moment()],
+        'Last 30 Days': [moment().subtract(29, 'days'), moment()],
+        'This Month': [moment().startOf('month'), moment()],
+        'Last Month': [moment().subtract(1, 'month').startOf('month'), moment().subtract(1, 'month').endOf('month')],
+        'Last 5 minutes': [moment().subtract(5, 'minutes'), moment()],
+        'Last 15 minutes': [moment().subtract(15, 'minutes'), moment()],
+        'Last 30 minutes': [moment().subtract(30, 'minutes'), moment()],
+        'Last 1 hour': [moment().subtract(1, 'hours'), moment()],
+        'Last 3 hours': [moment().subtract(3, 'hours'), moment()],
+        'Last 6 hours': [moment().subtract(6, 'hours'), moment()],
+        'Last 12 hours': [moment().subtract(12, 'hours'), moment()],
+        'Last 24 hours': [moment().subtract(24, 'hours'), moment()],
+        'Last 2 days': [moment().subtract(2, 'days'), moment()],
+        'Last 7 days': [moment().subtract(7, 'days'), moment()],
+        'Last 30 days': [moment().subtract(30, 'days'), moment()],
+        'Last 90 days': [moment().subtract(90, 'days'), moment()],
+        'Last 6 months': [moment().subtract(6, 'months'), moment()],
+        'Last 1 year': [moment().subtract(1, 'years'), moment()],
+        'Today': [moment().startOf('day'), moment()],
+        'Yesterday': [moment().subtract(1, 'days').startOf('day'), moment().startOf('day')],
+        'This week': [moment().startOf('week'), moment()],
+        'This month': [moment().startOf('month'), moment()],
+        'Last month': [moment().subtract(1, 'month').startOf('month'), moment().subtract(1, 'month').endOf('month')]
```

Comment:
As long as we don't support a `startOf` operator for relative timestamp in the URL, this should not be in the list. Unfortunately with the current heuristic, I see no way to support `startOf`.